test/models/identity/roles_tests.rb

```diff
@@ -0,0 +1,28 @@
+require "test_helper"
+
+describe "Fog::Identity[:openstack] | roles" do
+  before do
+    @identity = Fog::Identity[:openstack]
+
+    @tenant   = @identity.tenants.create(:name => 'test_user')
+    @user     = @identity.users.create(:name => 'test_user', :tenant_id => @tenant.id, :password => 'spoof')
+    @role     = @identity.roles(:user => @user, :tenant => @tenant).create(:name => 'test_role')
+    @roles    = @identity.roles(:user => @user, :tenant => @tenant)
+  end
+
+  describe "success" do
+    it "#all" do
+      @roles.all.must_be_kind_of Fog::Identity::OpenStack::V2::Roles
+    end
+
+    it "#get" do
+      @roles.get(@roles.first.id).body.wont_be_nil
+    end
+
+    after do
```

Comment:
I would move this `after` block in the beginning of appropriate `describe` block or maybe even one level upper here to the same level as `before` can be found.